api/src/helpers/origin.ts

```diff
@@ -0,0 +1,36 @@
+import { GraphQLError } from 'graphql';
+import wildcard from 'wildcard-match';
+
+export function isOriginValid(originParam: string): boolean {
+  if (originParam === '*') {
+    return true;
+  }
+  if (originParam.includes(' ')) {
+    return false;
+  }
+  const origins = originParam.split(',');
+  return origins.some((origin) => {
+    if (!origin.startsWith('http://') && !origin.startsWith('https://')) {
+      return false;
+    }
+    return true;
+  });
+}
+export function throwGQLErrorIfOriginDisallowed(
+  originRestriction: string,
+  originReceived?: string,
+) {
+  if (originRestriction !== '*') {
```

Comment:
Please prefer to return earlier
```suggestion
  if (originRestriction === '*') {
      return;
  }
```